content/post/new-python-api.md

```diff
@@ -0,0 +1,263 @@
+
+```python
+@wallaroo.partition
+def partition(data):
+    if data[0] >= 'a' or data[0] <= 'z':
+        return data[0]
+    else:
+        return "!"
+```
+
+#### Incoming Messages and the Decoder
+
+The `decoder` contains the logic for interpreting incoming bytes from a TCP stream into an object that represents the message within the application. In this example, incoming messages are represented as strings.
+
+```python
+@wallaroo.decoder(header_length=4, length_fmt=">I")
+def decoder(bs):
+    return bs.decode("utf-8")
+```
+
+The `@wallaroo.decoder` decorator must be used to indicate that this is a decoder. The `header_length` argument specifies the number of bytes that will be used for the message length. The `length_fmt` argument specifies the way the length bytes are structured using the same format that is used by the `struct` module. In this case, `">I"` means that the length will be a big-endian 32-bit integer.
+
+#### Stateless Computation
+
+`split` is a stateless computation. It takes a string and splits it into a list of strings where each string in the list represents a word.
+
+```
+"why hello world" -> Split -> ["why", "hello", "world"]
+```
+
+Here's what the `split` computation looks like:
+
+```python
+@wallaroo.computation_multi(name="split into words")
+def split(data):
+    punctuation = " !\"#$%&'()*+,-./:;<=>?@[\]^_`{|}~"
+
+    words = []
+
+    for line in data.split("\n"):
+        clean_line = line.lower().strip(punctuation)
+        for word in clean_line.split(' '):
+            clean_word = word.strip(punctuation)
+            words.append(clean_word)
+
+    return words
+```
+
+The `split` computation returns a list of individual words that the Wallaroo framework sends along as messages to the next step in the pipeline. Wallaroo takes care of making sure that each message gets delivered to the correct partition. Your application does not need to know how the data is partitioned or which machine holds that partition. The `@wallaroo.computation_multi(...)` decorator must be used to indicate that this is a computation that returns multiple outgoing messages. The `name` argument specifies the name of the computation that will be used by Wallaroo when reporting information about the application.
+
+#### State Computation
+
+`count_word` is a state computation; it uses an incoming message and a state entity to update the word count for the new word and returns a message for Wallaroo to send on its behalf.
+
+```python
+@wallaroo.state_computation(name="Count Word")
+def count_word(word, word_totals):
+    word_totals.update(word)
+    return (word_totals.get_count(word), True)
+```
+
+The `@wallaroo.state_computation(...)` decorator must be used to indicate that this is a state computation. As with the computation above, the `name` argument specifies the name that Wallaroo will use when reporting information about the application.
```

Comment:
It might be useful to mention why the `True` is returned.

Reply:
I'll add a clarification.